From d1ee0859b400e6d88670413d24021f9b229aae92 Mon Sep 17 00:00:00 2001
From: chunrong <guochunrong@phytium.com.cn>
Date: Tue, 5 Jan 2021 11:23:39 +0800
Subject: [PATCH]  fix platform

Signed-off-by: chunrong <guochunrong@phytium.com.cn>
---
 pyanaconda/platform.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyanaconda/platform.py b/pyanaconda/platform.py
index 3094e8cf9..64424bea2 100644
--- a/pyanaconda/platform.py
+++ b/pyanaconda/platform.py
@@ -292,8 +292,8 @@ def get_platform():
             return EFI()
     elif arch.is_x86():
         return X86()
-    elif arch.is_arm():
-        return ARM()
+    elif arch.is_aarch64():
+        return Aarch64EFI()
     else:
         raise SystemError("Could not determine system architecture.")
 
-- 
2.17.1
