diff --git a/drivers/irqchip/irq-gic-v3-its-pci-msi.c b/drivers/irqchip/irq-gic-v3-its-pci-msi.c
index 8d6d009d1..6bb0db162 100644
--- a/drivers/irqchip/irq-gic-v3-its-pci-msi.c
+++ b/drivers/irqchip/irq-gic-v3-its-pci-msi.c
@@ -39,6 +39,9 @@ static struct irq_chip its_msi_irq_chip = {
 	.irq_mask		= its_mask_msi_irq,
 	.irq_eoi		= irq_chip_eoi_parent,
 	.irq_write_msi_msg	= pci_msi_domain_write_msg,
+#ifdef CONFIG_IPIPE
+	.flags			= IRQCHIP_PIPELINE_SAFE,
+#endif
 };
 
 static int its_pci_msi_vec_count(struct pci_dev *pdev, void *data)
diff --git a/drivers/irqchip/irq-gic-v3-its-platform-msi.c b/drivers/irqchip/irq-gic-v3-its-platform-msi.c
index 7b8e87b49..8e6900f9e 100644
--- a/drivers/irqchip/irq-gic-v3-its-platform-msi.c
+++ b/drivers/irqchip/irq-gic-v3-its-platform-msi.c
@@ -23,6 +23,9 @@
 
 static struct irq_chip its_pmsi_irq_chip = {
 	.name			= "ITS-pMSI",
+#ifdef CONFIG_IPIPE
+	.flags			= IRQCHIP_PIPELINE_SAFE,
+#endif
 };
 
 static int of_pmsi_get_dev_id(struct irq_domain *domain, struct device *dev,
diff --git a/drivers/irqchip/irq-phytium-ixic.c b/drivers/irqchip/irq-phytium-ixic.c
index dd3d755b7..5746f40a7 100644
--- a/drivers/irqchip/irq-phytium-ixic.c
+++ b/drivers/irqchip/irq-phytium-ixic.c
@@ -53,12 +53,28 @@ static void phytium_ixic_irq_eoi(struct irq_data *d)
 	irq_chip_eoi_parent(d);
 }
 
+#ifdef CONFIG_IPIPE
+static void phytium_ixic_hold_irq(struct irq_data *d)
+{
+	return;
+}
+
+static void phytium_ixic_release_irq(struct irq_data *d)
+{
+	phytium_ixic_irq_eoi(d);
+}
+#endif
+
 static struct irq_chip phytium_ixic_irq_chip = {
 	.name = "IXIU",
 	.irq_eoi = phytium_ixic_irq_eoi,
 	.irq_mask = irq_chip_mask_parent,
 	.irq_unmask = irq_chip_unmask_parent,
 	.irq_set_type = irq_chip_set_type_parent,
+#ifdef CONFIG_IPIPE
+	.irq_hold = phytium_ixic_hold_irq,
+	.irq_release = phytium_ixic_release_irq,
+#endif
 	.irq_set_affinity = irq_chip_set_affinity_parent,
 	.flags = IRQCHIP_MASK_ON_SUSPEND,
 };
